qiskit_machine_learning/runtime/torch_runtime_client.py

```diff
@@ -0,0 +1,513 @@
+                        ("classification" for classification, and "regression" for regression)
+                        or a custom scoring function defined as:
+                        ``def score_func(model_output, target): -> score: float``.
+        Returns:
+            score: A metric of the model's performance.
+        Raises:
+            ValueError: If the backend has not yet been set.
+            ValueError: If the provider has not yet been set.
+            ValueError: If "score_func" is not "classification", "regression",
+                or a custom scoring function.
+            RuntimeError: If the job execution failed.
+        """
+        if self._backend is None:
+            raise ValueError("The backend has not been set.")
+
+        if self._provider is None:
+            raise ValueError("The provider has not been set.")
+
+        # serialize model using pickle + dill
+        serial_model = obj_to_str(self._model)
+
+        if score_func == "classification":
+            self._score_func = _score_classification
+        elif score_func == "regression":
+            self._score_func = MSELoss()
+        elif callable(score_func):
+            self._score_func = score_func
+        else:
+            raise ValueError(
+                '"score_func" must be a string for the available scoring functions',
+                '("classification" or "regression"), or a custom scoring function.',
+            )
+
+        # serialize loss function using pickle + dill
+        serial_score_func = obj_to_str(self._score_func)
+        serial_data = obj_to_str(data_loader)
+
+        # define runtime options
+        options = {"backend_name": self._backend.name()}
+
+        # combine the settings with the serialized buffers to runtime inputs
+        inputs = {
+            "model": serial_model,
+            "data": serial_data,
+            "score_func": serial_score_func,
+            "shots": self._shots,
+            "measurement_error_mitigation": self.measurement_error_mitigation,
+        }
+
+        # send job to runtime and return result
+        job = self.provider.runtime.run(
+            program_id="torch-infer",
+            inputs=inputs,
+            options=options,
+        )
+
+        # Raise a runtime error with the job id if something goes wrong
+        try:
+            result = job.result()
+        except Exception as exc:
+            raise RuntimeError(f"The job {job.job_id()} failed unexpectedly.") from exc
+
+        return result["score"]
+
+
+def obj_to_str(obj: Any) -> str:
+    """
+    Encodes any object into a JSON-compatible string using dill. The intermediate
+    binary data must be converted to base 64 to be able to decode into utf-8 format.
+
+    Returns:
+        The encoded string
+    """
+    string = base64.b64encode(dill.dumps(obj, byref=False)).decode("utf-8")
+    return string
+
+
+def str_to_obj(string: str) -> Any:
```

Comment:
Do we need them as public functions?

Reply:
please see the above comment for `qiskit_machine_learning/runtime/__init__.py`